Complete ticket 7

roseymars opened the pull request:
Complete ticket 7 PATCH /api/articles/:article_id

- tested for both incremented and decremented votes
- implemented checkExists into addArticleVotes (added checkExists to utils.js)
- added a handlingServerErrors function to handle status 500

roseymars committed:
minor refactor, removed promise reject

roseymars committed:
complete ticket-7

dav3rid commented on __tests__/app.test.js:
```diff
@@ -52,18 +52,89 @@ describe("GET /api/articles/article_id", () => {
+    return request(app)
+      .patch("/api/articles/7")
+      .send({ inc_votes: 50 })
+      .expect(201)
```
I would actually suggest a good old 200 is actually more appropriate here. 201 "Created" is more often associated with the creation of a brand new resource rather than an amendment to an existing one

dav3rid commented on __tests__/app.test.js:
```diff
@@ -52,18 +52,89 @@ describe("GET /api/articles/article_id", () => {
+      .expect(404)
+      .then(({ body: { msg } }) => {
+        expect(msg).toBe("PATH REQUESTED NOT FOUND");
+      });
```
Another potential 400 test here could be for missing the `inc_votes` key entirely on the request body

dav3rid commented on app.js:
```diff
@@ -11,14 +11,16 @@ app.all('/api', (req, res) => {
 
 app.all('/*', (req, res) => {
-    res.status(404).send( { msg: "PATH REQUESTED NOT FOUND" } )
+    res.status(404).send(( { msg: "PATH REQUESTED NOT FOUND" } ))
```
Not hurting anyone - but you shouldn't need the additional layer of parentheses here

dav3rid commented on db/helpers/utils.js:
```diff
@@ -20,3 +24,15 @@ exports.formatComments = (comments, idLookup) => {
+    return Promise.reject({ status: 404, msg: 'Resource not found' });
+  }
+})
+};
```
Purely an opinion - but I might suggest this new function exist in another file away from the seeding util functions - maybe a models/utils.js?
Also, wonderful generalised function here!

dav3rid commented on errors/index.js:
```diff
@@ -1,6 +1,6 @@
-  if (err.status && err.msg) {
-    res.status(err.status).send({ msg: err.msg });
+  if (err.status === 404 && err.msg === "Resource not found") {
+    res.status(404).send({ msg: "PATH REQUESTED NOT FOUND" });
```
I think the previous setup may have been preferable here - remember that for this error handler, you have created the custom error object anyway so should be able to reuse elsewhere

dav3rid commented on errors/index.js:
```diff
@@ -10,6 +10,10 @@ exports.handlePsqlErrors = (err, req, res, next) => {
+
+exports.handleServerErrors = (err, req, res, next) => {
+  if (err.status === 500) res.status(500).send( { msg: 'Internal Server Error'})
+}
```
For this one - you can simply hardcode a status of 500 without the conditional logic
If you even get to this error handler, it hasn't already been handled by one above.
Also, a great place to console.log the error as it's one you won't have come across yet!

roseymars committed:
amended code for ticket-7 based on comments

roseymars commented:
I have amended code based on Dave's wonderful suggestions:

- added another sad path (400) test for when inc_votes key is missing
- created a new file within models (models-utils.js) for my checkExists function
- removed conditional logic in errors/index.js when handling 500s
- removed stray layer of parentheses from app.all
- changed status code to 200 for my happy paths

dav3rid reviewed (approved):
👍
